irohad/multi_sig_transactions/transport/impl/mst_transport_grpc.cpp

```diff
@@ -140,30 +144,43 @@ void MstTransportGrpc::subscribe(
   subscriber_ = notification;
 }
 
-void MstTransportGrpc::sendState(const shared_model::interface::Peer &to,
-                                 ConstRefState providing_state) {
-  log_->info("Propagate MstState to peer {}", to.address());
-  sendStateAsyncImpl(to,
-                     providing_state,
-                     my_key_,
-                     *async_call_,
-                     sender_factory_.value_or(default_sender_factory));
+rxcpp::observable<bool> MstTransportGrpc::sendState(
+    shared_model::interface::Peer const &to, MstState const &providing_state) {
+  return rxcpp::observable<>::create<bool>([&](auto s) {
+    log_->info("Propagate MstState to peer {}", to.address());
+    sendStateAsyncImpl(to,
+                       providing_state,
+                       my_key_,
+                       *async_call_,
+                       [s](auto &status, auto &) {
+                         s.on_next(status.ok());
+                         s.on_completed();
+                       },
+                       sender_factory_.value_or(default_sender_factory));
```

Comment:
```suggestion
                       },
                       sender_factory_.value_or(default_sender_factory));
    s.on_completed();
```

Reply:
Discussed offline, `on_completed` should not be called right after making an asynchronous call.